src/Payum/Paypal/ProHosted/Nvp/Action/Api/GetTransactionDetailsAction.php

```diff
@@ -0,0 +1,42 @@
+<?php
+namespace Payum\Paypal\ProHosted\Nvp\Action\Api;
+
+use Payum\Core\Bridge\Spl\ArrayObject;
+use Payum\Core\Exception\LogicException;
+use Payum\Paypal\ProHosted\Nvp\Api;
+use Payum\Paypal\ProHosted\Nvp\Request\Api\GetTransactionDetails;
+use Payum\Core\Exception\RequestNotSupportedException;
+
+class GetTransactionDetailsAction extends BaseApiAwareAction
+{
+    /**
+     * {@inheritDoc}
+     */
+    public function execute($request)
+    {
+        /** @var $request GetTransactionDetails */
+        RequestNotSupportedException::assertSupports($this, $request);
+        $model = ArrayObject::ensureArrayObject($request->getModel());
+
+        $fields                  = new ArrayObject([]);
+        $fields['TRANSACTIONID'] = $model['txn_id'];
+
+        if (null == $fields['TRANSACTIONID']) {
+            throw new LogicException('TRANSACTIONID must be set.');
```

Comment:
it would be good to base an exception on a model, so here you should check that $model['txn_id'] must be set.